Feat/new listing v2

CJ-Rose opened the pull request:
@thebeyondr i created this PR from your branch so i can view these commits all together. there is a lot here for me to unpack... just trying to keep up 😅

thebeyondr committed:
feat: :heavy_plus_sign: Add `tailwindcss`

thebeyondr committed:
chore: :construction_worker: Setup tailwind with webpack

thebeyondr committed:
feat: :heavy_plus_sign: Add `@shadcn/ui`

thebeyondr committed:
feat: :lipstick: Add `shadcn/ui` dialog primitive

thebeyondr committed:
feat: :construction: In progress `NewListing` dialog using tailwindcss

thebeyondr committed:
feat: :construction: In progress `NewListing` page

Add working state and reducers for form steps

thebeyondr committed:
feat: :sparkles: Add `General Info` step to new listing

thebeyondr committed:
chore: :lipstick: Add toggle component

thebeyondr committed:
feat: :sparkles: Add TrustScore step section to New Listing

Remove frequency from input since its always one time, refactor form header and footer into their own components

thebeyondr committed:
chore: :heavy_plus_sign: Add form related packges

Added `zod`,  `label` and `form` radix primitives, `react-hook-form`

thebeyondr committed:
refactor: :construction: Update NewListing form to use zod and react-hook-forms

Selecting categories now works!!! 💥💥

thebeyondr committed:
refactor: :recycle: Convert `GeneralInfo` section to a controlled `react-hook-form` component

General Info section are now all controlled inputs + use shadcn UI, convert price to dollar amount, reintegrate footer

thebeyondr committed:
chore: :heavy_plus_sign: Add `Checkbox` primitive

thebeyondr committed:
refactor: :recycle: Turn `SelectCategoryFormStep` into a controlled component using `react-hook-forms`

Remove `selectedCategories` state and other unused code as well

thebeyondr committed:
chore: :heavy_minus_sign: Remove `Toggle` primitive & component

thebeyondr committed:
chore: :heavy_plus_sign: Add `Switch` component

thebeyondr committed:
refactor: :recycle: Update `TrustScoreFormStep` to be controlled Switches

Add `TrustScoreKeyEnum`

thebeyondr committed:
refactor: :recycle: Update `TrustScore` step

Dynamically create inital categories based on `listingTypes`, change `listingCategories` to be an object with sections as keys

thebeyondr committed:
refactor: :lipstick: Re-add form header

Remove unused reducer files, add some todos

thebeyondr committed:
refactor: :lipstick: Update `NewListing` page

Add todos, update form styles

thebeyondr committed:
chore: :fire: Remove `NewListing` modal

thebeyondr committed:
refactor: :recycle: Add `Create Listing` link to `Header`

thebeyondr committed:
refactor: :lipstick: Update `NewListing` styles

Increase text sizes, Add boxes around trust scores, update footer styles

thebeyondr committed:
refactor: :truck: Move some types and constants from `NewListing` into their own files

thebeyondr committed:
feat: :sparkles: Add `useTrustlist` hook

thebeyondr committed:
feat: :heavy_plus_sign: Add `RadioGroup`

thebeyondr committed:
chore: :fire: Archive `Dashboard` page

thebeyondr committed:
chore: :truck: Remove Dashboard page from index

thebeyondr committed:
fix: :speech_balloon: Remove all categories except devconnect from `listingCategories`

thebeyondr committed:
feat: :sparkles: Update NewListing form

category selection uses radio inputs now, we validate form fields before going to the next step, add some todos

thebeyondr committed:
chore: :memo: Add todos to `useTrustlist` hook

thebeyondr committed:
style: :adhesive_bandage: Formatting and removing unused code

CJ-Rose committed:
display contract data as scores for new listing

thebeyondr committed:
chore: :package: Update `yarn.lock`

thebeyondr committed:
fix: :bug: Fix scores being NaN

Update `useEffect`: Escape if provableData is empty, update object already used, remove exhaustive prop

CJ-Rose committed:
remove comments

CJ-Rose committed:
Merge pull request #48 from trustlist/display-scores

get scores for new listing

CJ-Rose committed:
add epoch info, change timer

CJ-Rose committed:
add epoch info text

CJ-Rose committed:
update header button layout/style

CJ-Rose committed:
remove pencil svg

CJ-Rose committed:
make sections/categories blue again

CJ-Rose committed:
try to delete pencil again

CJ-Rose committed:
add toast to actions

thebeyondr committed:
Merge pull request #49 from trustlist/header/epoch-info

header/epoch-info

thebeyondr committed:
feat: :sparkles: Add `epochKey` and `posterId` to form before publishing

Transition the user to the new epoch if they are behind

CJ-Rose commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,421 @@
+    frequency: 'one time',
+    description: '',
+    posterId: '',
+    revealTrustScores: {
+      [TrustScoreKeyEnum.LP]: true,
+      [TrustScoreKeyEnum.LO]: true,
+      [TrustScoreKeyEnum.CB]: true,
+      [TrustScoreKeyEnum.GV]: true,
+    },
+    // scores: {
+    //   [TrustScoreKeyEnum.LP]: 0.3,
+    //   [TrustScoreKeyEnum.LO]: 0.2,
+    //   [TrustScoreKeyEnum.CB]: 0.03,
+    //   [TrustScoreKeyEnum.GV]: 0.5,
+    // }
```
i think we can remove `revealTrustScores` from the listing (and `isRevealed` from the record in `data`)
.. bc we shouldn't store scores that the user wants to hide. the idea is for the listing to show a (chosen to be) revealed score, or that the score is not being revealed, or that the user has not accumulated any data for that metric. (bc zero data is a different thing than a 0% score).
so i'd like to just add the score as a string, it can be either the score, or an X if hidden, or n/a if no data.
.. next comment 👇

CJ-Rose commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,421 @@
+          render={({ field }) => (
+            <FormItem className='flex justify-between items-start space-x-6 border border-muted-foreground p-3' key={key}>
+              <div>
+                <FormLabel className="text-foreground text-lg" htmlFor={key}>
+                  {scoreInfo.title}:{' '}<span className="text-blue-700 font-extrabold">{scoreInfo.score}%</span>
+                </FormLabel>
+                <FormDescription className='text-foreground/80'>{scoreInfo.description}</FormDescription>
+              </div>
+              <FormControl>
+                <div className="flex space-x-2">
+                  <Switch
+                    id={key}
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                    className={cn(field.value ? 'data-[state=checked]:bg-blue-700' : '')}
+                  />
+                  <p className='text-muted-foreground'>{field.value ? 'Show' : 'Hide'}</p>
+                </div>
+              </FormControl>
```
how can we use the switch to change the stored score to an X if the user chooses to hide?

if we can send like scores = {LP : 45, LO: 89, CB: X, GV: n/a} as a string to the database, i think i can figure out how to read the scores into the other components that display them, without having to rewrite too much. also i think i will be able to get rid of that second crazy math function in the Trustlist hook 😂

thebeyondr replied:
I don't think we should remove `revealTrustScores`. We can just use it to determine the actual scores we'll send to the API in  the `publishListing` fn

 Example:
Let's say the user chose to reveal LP and LO, then we'd use `revealTrustScores` to figure out the scores sent to api: `{LP : 45, LO: 89}`

Even if we send `revealTrustscores` in the API request,  the backend is not setup to do anything with it so it won't affect the db

ChialiT committed:
Update config.ts

push to make vercel work, will roll back

ChialiT committed:
Update config.ts

revert changes

thebeyondr committed:
refactor: :recycle: Update `trustScore` calculation

make `trustScoresFromData` a state variable and initialize score values to `undefined`

thebeyondr committed:
feat: :speech_balloon: Update data being sent to create listing

thebeyondr committed:
chore: :heavy_plus_sign: Add `@unirep-app/contracts`

thebeyondr committed:
refactor: :heavy_plus_sign: add `@unirep-app/circuits` to `peerDependencies`

CJ-Rose committed:
give user rep data for posting

CJ-Rose committed:
add scores to data, send to db

CJ-Rose committed:
fix score display on listings and listing detail

CJ-Rose commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,445 @@
+              </div>
+              <FormControl>
+                <div className="flex space-x-2">
+                  <Switch
+                    id={key}
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                    className={cn(field.value ? 'data-[state=checked]:bg-blue-700' : '')}
+                  />
```
@thebeyondr the Switch doesn't appear to change `isRevealed` to false. when i choose hide a scores from the ui,  are still sent on the listing. i don't understand where `field.value` and `field.onChange` come from, where are we asking `isRevealed` to flip from T/F ?
the logic to create score obj to send to the db (only add score if `isRevealed`) is next comment 👇

thebeyondr replied:
Yeah it doesn't use `isRevealed`. its connected to `revealTrustScores` (see line 246)

CJ-Rose commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,445 @@
+        throw new Error("Failed to get epoch and key");
+      }
+      const { currentEpoch, userEpochKey, nonce } = epochAndKey;
+      for (let i=0; i<4; i++) {
+        if (trustScoresFromData[trustScoreKeys[i]].isRevealed) {
+          data.scores[trustScoreKeys[i]] = trustScoresFromData[trustScoreKeys[i]].score
+        }
+      }
```
here

thebeyondr replied:
will update

thebeyondr committed:
fix: :fire: Remove `isRevealed` and calculate scores for new listing

thebeyondr commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,445 @@
+          amount: String(data.price),
+          amountType: data.frequency,
+          scoreString: JSON.stringify(data.scores)
+        }
```
@CJ-Rose the backend will probably nee dto be updated to handle this new format if it doesn't yet

thebeyondr commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,445 @@
+    return { currentEpoch: currentEpoch, userEpochKey: posterId, nonce: epkNonce }
+  }
+
+  const publishPost = async (data: ListingFormValues) => {
```
@CJ-Rose This function takes the `revealTrustScores` from the toggle inputs and churns out the scores (or thats what its supposed to do lol)

thebeyondr replied:
@CJ-Rose This function takes the `revealTrustScores` from the toggle inputs and churns out the scores (or thats what its supposed to do lol

thebeyondr committed:
refactor: :recycle: More readable names for variables in `generateScores`

thebeyondr commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,449 @@
+    return { currentEpoch: currentEpoch, userEpochKey: posterId, nonce: epkNonce }
+  }
+
+const generateScores = (scoresRevealed: Record<TrustScoreKey, boolean>) => {
```
@CJ-Rose This function takes the `revealTrustScores` from the toggle inputs and churns out the scores (or thats what its supposed to do lol)

thebeyondr commented on packages/frontend/src/pages/NewListing.tsx:
```diff
@@ -0,0 +1,449 @@
+          epoch: currentEpoch,
+          category: Object.values(data.categories)[0],
+          section: Object.keys(data.categories)[0],
+          posterId: userEpochKey,
```
@CJ-Rose backend will need an update to handle this (the "stringified" object), right?

CJ-Rose committed:
change score type to string

CJ-Rose committed:
add contact info to newListing

CJ-Rose committed:
add ui messages on auto transition and publish

CJ-Rose committed:
fix styles on Listings component

CJ-Rose committed:
fix detail modal style

CJ-Rose committed:
NewOffer page (not done)

CJ-Rose committed:
NewOffer page (done)

CJ-Rose committed:
update offer success message

CJ-Rose committed:
DetailModal- update offers and add toast

CJ-Rose committed:
DetailModal details

vivianjeng committed:
chore: update docker file

CJ-Rose committed:
Merge pull request #51 from trustlist/fix-docker

chore: update docker file

vivianjeng committed:
fix(relay): fix tx manager

CJ-Rose committed:
scores are numbers, need to update scores sent to db

CJ-Rose committed:
generate a new ID if it is empty

CJ-Rose committed:
Merge branch 'feat/new-listing-v2' of https://github.com/trustlist/trustlist into feat/new-listing-v2

CJ-Rose committed:
Merge pull request #54 from trustlist/fix-tx-manager

fix(relay): fix tx manager

CJ-Rose committed:
Merge branch 'feat/new-listing-v2' of https://github.com/trustlist/trustlist into scores-are-numbers

CJ-Rose committed:
use raw scores before calc

CJ-Rose committed:
remove tooltip from Listings

CJ-Rose committed:
update Listings+DetailModals bc scores are numbers

CJ-Rose committed:
remove unused code

CJ-Rose committed:
delete db

vivianjeng committed:
fix(frontend): fix env file

vivianjeng committed:
Merge remote-tracking branch 'origin' into feat/new-listing-v2

vivianjeng committed:
Merge branch 'feat/new-listing-v2' into deployment

CJ-Rose committed:
fix names of score parts

CJ-Rose committed:
Merge pull request #55 from trustlist/scores-are-numbers

Scores are numbers

CJ-Rose committed:
Merge pull request #56 from trustlist/deployment

Fix webpack.config.js file

CJ-Rose committed:
get proveable score for offer

CJ-Rose committed:
Merge branch 'feat/new-listing-v2' of https://github.com/trustlist/trustlist into feat/new-listing-v2

CJ-Rose committed:
prevent offers on own listing and cleam up comments
